ext/js/dictionary/dictionary-data-util.js

```diff
@@ -82,18 +82,57 @@ export function groupTermFrequencies(dictionaryEntry) {
     }
 
     const results = [];
+
+    /** @type {import('dictionary').AverageFrequencyListTerm} */
+    const averages = {};
     for (const [dictionary, map2] of map1.entries()) {
         const frequencies = [];
         const dictionaryAlias = aliasMap.get(dictionary) ?? dictionary;
-        for (const {term, reading, values} of map2.values()) {
+        for (let {term, reading, values} of map2.values()) {
             frequencies.push({
                 term,
                 reading,
                 values: [...values.values()],
             });
+            const valuesArray = [...values.values()];
+            if (reading === null) { reading = ''; }
+            let {currentAvg, count} = averages[term]?.[reading] ?? {currentAvg: 1, count: 0};
+            if (valuesArray[0].frequency === null) { continue; }
+
+            currentAvg = (count / (currentAvg)) + (1 / (valuesArray[0].frequency));
+            currentAvg = (count + 1) / currentAvg;
+            count += 1;
+
+            averages[term] = {
+                ...averages[term],
+                [reading]: {
+                    currentAvg, count,
+                },
+            };
         }
         results.push({dictionary, frequencies, dictionaryAlias});
     }
+
+    for (const currentTerm of Object.keys(averages)) {
```

Comment:
Should averages really be defined as an object if you need to loop over it here?